src/main/java/picard/util/IntervalList/IntervalListScattererByIntervalCount.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.util.IntervalList;
+
+import htsjdk.samtools.util.CollectionUtil;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+
+import java.util.List;
+
+/**
+ * Scatters {@link IntervalList} by interval count so that resulting {@link IntervalList}'s have the same number of intervals in them.
+ * An exception is made for the final interval list, which will always contain at least this number of intervals,
+ * but may also contain the remainder.
+ */
+public class IntervalListScattererByIntervalCount implements IntervalListScatterer {
+
+    @Override
+    public long intervalWeight(final Interval interval) {
+        return 1;
+    }
+
+    @Override
+    public long listWeight(final IntervalList intervalList) {
+        return intervalList.size();
+    }
+
+    @Override
+    public List<Interval> takeSome(final Interval interval, final long idealSplitWeight, final long currentSize, final double projectSizeOfRemaining) {
+        final long amount = idealSplitWeight - currentSize;
+        if (amount > 0) {
+            return CollectionUtil.makeList(interval, null);
```

Comment:
Why not use `Arrays.asList()` here? It looks like the result of this method is never modified.

Reply:
I'm not sure I understand the difference between using makeList versus asList...

Reply:
The basic difference is that `CollectionUtil.makeList()` returns a modifiable list and `Arrays.asList()` returns an unmodifiable one.

Also, `Arrays` is part of the Java API, and `CollectionUtil` is part of htsjdk. In general I think it's better to use a Java API when possible.